fe/fe-core/src/main/java/org/apache/doris/analysis/CreateSqlBlockRuleStmt.java

```diff
@@ -116,6 +118,11 @@ public void analyze(Analyzer analyzer) throws UserException {
         // check properties
         CreateSqlBlockRuleStmt.checkCommonProperties(properties);
         setProperties(properties);
+
+        // avoid a rule block any ddl for itself
+        if (StringUtils.isNotEmpty(sql) && Pattern.compile(sql).matcher(this.ruleName).find()) {
```

Comment:
why using `sql` to match the `ruleName`?

Reply:
if a sql_block_rule named block_rule_delete with sql "delete" is created, then any update to this block_rule will be blocked by itself, like "DROP SQL_BLOCK_RULE block_rule_delete", coz sql block rule name is a part of sql now.